### PR TITLE
[FANCTRL] Fix enums

### LINKED ISSUE #33857: [1.3] All values of FanModeSequenceEnum in FanControl cluster are Off
### Reproduction steps

In this commit, all values in the `FanModeSequenceEnum` got changed to `Off`. I don’t think it is intentional and seems like a bug.

https://github.com/project-chip/connectedhomeip/commit/9c0cd9eb3656d3c7e40ca811257270db50384d61#diff-5e76a5680a663888acfb0e00e8da9465bfe3ab6f98a92e7a05248fc80abe2401

### Bug prevalence

Haven't tested, but should be persistent

### GitHub hash of the SDK that was being used

Not tested

### Platform

other

### Platform Version(s)

_No response_

### Type

Spec Compliance Issue

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by rbultman**
This PR removes the slashes in the enum names for fan control.

Spec PR here: https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/9369

Fixes #33857

### CODE REVIEW

File: data_model/clusters/FanControl.xml [hunk 1]
```diff
@@ -119,36 +119,36 @@ Davis, CA 95616, USA
       </item>
     </enum>
     <enum name="FanModeSequenceEnum">
-      <item value="0" name="Off" summary="Fan is capable of off, low, medium and high modes">
+      <item value="0" name="OffLowMediumHigh" summary="Fan is capable of off, low, medium and high modes">
```
Comment (andy31415) [inline data_model/clusters/FanControl.xml:122]: These files are to be modified by the XML data model parser. We should not update them manually.

Comment (rbultman) [inline data_model/clusters/FanControl.xml:122]:
```suggestion
      <item value="0" name="Off" summary="Fan is capable of off, low, medium and high modes">
```
```suggestion
      <item value="0" name="OffLowMediumHigh" summary="Fan is capable of off, low, medium and high modes">
```

File: data_model/clusters/FanControl.xml [hunk 2]
```diff
@@ -126,29 +126,29 @@ Davis, CA 95616, USA
           </notTerm>
         </optionalConform>
       </item>
-      <item value="1" name="Off" summary="Fan is capable of off, low and high modes">
+      <item value="1" name="OffLowHighAuto" summary="Fan is capable of off, low and high modes">
```
Comment (rbultman) [inline data_model/clusters/FanControl.xml:129]:
```suggestion
      <item value="1" name="Off" summary="Fan is capable of off, low and high modes">
```

File: data_model/clusters/FanControl.xml [hunk 3]
```diff
@@ -126,29 +126,29 @@ Davis, CA 95616, USA
...
         <optionalConform choice="a">
           <notTerm>
             <feature name="AUT"/>
           </notTerm>
         </optionalConform>
       </item>
-      <item value="2" name="Off" summary="Fan is capable of off, low, medium, high and auto modes">
+      <item value="2" name="OffLowMediumHighAuto" summary="Fan is capable of off, low, medium, high and auto modes">
```
Comment (rbultman) [inline data_model/clusters/FanControl.xml:136]:
```suggestion
      <item value="2" name="Off" summary="Fan is capable of off, low, medium, high and auto modes">
```

File: data_model/clusters/FanControl.xml [hunk 4]
```diff
@@ -126,29 +126,29 @@ Davis, CA 95616, USA
...
         <optionalConform choice="a">
           <feature name="AUT"/>
         </optionalConform>
       </item>
-      <item value="3" name="Off" summary="Fan is capable of off, low, high and auto modes">
+      <item value="3" name="OffLowHighAuto" summary="Fan is capable of off, low, high and auto modes">
```
Comment (rbultman) [inline data_model/clusters/FanControl.xml:141]:
```suggestion
      <item value="3" name="Off" summary="Fan is capable of off, low, high and auto modes">
```

File: data_model/clusters/FanControl.xml [hunk 5]
```diff
@@ -126,29 +126,29 @@ Davis, CA 95616, USA
...
         <optionalConform choice="a">
           <feature name="AUT"/>
         </optionalConform>
       </item>
-      <item value="4" name="Off" summary="Fan is capable of off, high and auto modes">
+      <item value="4" name="OffHighAuto" summary="Fan is capable of off, high and auto modes">
```
Comment (rbultman) [inline data_model/clusters/FanControl.xml:146]:
```suggestion
      <item value="4" name="Off" summary="Fan is capable of off, high and auto modes">
```

File: data_model/clusters/FanControl.xml (hunk 1 above)
Comment (rbultman) [reply data_model/clusters/FanControl.xml]: Reverted...

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Pull request merged**